service/schema/SchemaService.kt

```diff
@@ -341,6 +347,20 @@ class SchemaService(
 
     fun closeReadTx() = synchronized(this) { readTx.getAndSet(null)?.close() }
 
+    private fun resetLoadedConnectedTypes() {
+        for (typeState in entityTypes.values) typeState.resetLoadedConnectedTypes()
```

Comment:
Put `entityTypes.values`, `attributeTypes.values` etc. into a list or set and use `.forEach` to turn this (and `reloadLoadedConnectedTypes`) into a one-liner.

Reply:
Nice shout.
```kt
(entityTypes.values + attributeTypes.values + relationTypes.values).forEach { it.resetLoadedConnectedTypes() }
```

Reply:
That's a good start, but it would be much more elegant if we added `roleTypes` in as well!

I assume we can't because they don't have the right common base type. Why is this? Could we not define `resetLoadedConnectedTypes` (and `loadConstraints`) on `TypeState`? I assume it's currently defined on `ThingTypeState`, which means we can't reuse the API in `RoleTypeState`.